src/components/composer/standalone_composer/standalone_composer.ts

```diff
@@ -0,0 +1,92 @@
+import { Component } from "@odoo/owl";
+import { SELECTION_BORDER_COLOR } from "../../../constants";
+import { Store, useLocalStore, useStore } from "../../../store_engine";
+import { SpreadsheetChildEnv } from "../../../types/index";
+import { css, cssPropertiesToCss } from "../../helpers/css";
+import { ComposerSelection } from "../composer/cell_composer_store";
+import { Composer } from "../composer/composer";
+import { ComposerFocusStore, ComposerFocusType, ComposerInterface } from "../composer_focus_store";
+import { StandaloneComposerStore } from "./standalone_composer_store";
+
+css/* scss */ `
+  .o-spreadsheet {
+    .o-standalone-composer {
+      height: 28px;
+      overflow: auto;
+      box-sizing: border-box;
+
+      border-radius: 4px;
+      border: 1px solid;
+      border-color: #666666;
+
+      &.active {
+        border-color: ${SELECTION_BORDER_COLOR};
+      }
+
+      /* As the standalone composer is potentially very small (eg. in a side panel), we remove the scrollbar display */
+      scrollbar-width: none; /* Firefox */
+      &::-webkit-scrollbar {
+        display: none;
+      }
+    }
+  }
+`;
+
+interface Props {
+  onConfirm: (content: string) => void;
+  composerContent: string;
+  placeholder?: string;
+  class?: string;
+}
+
+export class StandaloneComposer extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-StandaloneComposer";
+  static props = {
+    composerContent: { type: String, optional: true },
+    onConfirm: Function,
+    placeholder: { type: String, optional: true },
+    class: { type: String, optional: true },
+  };
+  static components = { Composer };
+  static defaultProps = {
+    composerContent: "",
+  };
+
+  private composerFocusStore!: Store<ComposerFocusStore>;
+  private standaloneComposerStore!: Store<StandaloneComposerStore>;
+  private composerInterface!: ComposerInterface;
+
+  setup() {
+    this.composerFocusStore = useStore(ComposerFocusStore);
+    this.standaloneComposerStore = useLocalStore(StandaloneComposerStore, {
+      onConfirm: this.props.onConfirm,
```

Comment:
```suggestion
      onConfirm: (content) => this.props.onConfirm(content),
```
otherwise the callback is always the one from the first rendering. It's ok if the callback is a method on the component, but not if it's a closure such as:

```xml
<StandaloneComposer
  onConfirm="(content) => this.updateMeasureFormula(measure, content)"
  composerContent="measure.computedBy.formula"
  contextualAutocomplete="getMeasureAutocomplete(measure)"
/>
```
`measure` would always be the measure captured at the first rendering

Reply:
or even make `args` of `StandaloneComposerStore` a callback itself to be sure it gets the the latest version from the latest props?